drivers/gpio/gpio_ite_it8xxx2.c

```diff
@@ -0,0 +1,420 @@
+				gpio_port_pins_t mask, gpio_port_value_t value)
+{
+	const struct gpio_ite_cfg *gpio_config = DEV_GPIO_CFG(dev);
+	uint32_t port_val;
+
+	port_val = get_port(gpio_config);
+	port_val = (port_val & ~mask) | (value & mask);
+	set_port(gpio_config, port_val);
+	return 0;
+}
+
+static int gpio_ite_port_set_bits_raw(struct device *dev,
+				      gpio_port_pins_t pins)
+{
+	const struct gpio_ite_cfg *gpio_config = DEV_GPIO_CFG(dev);
+	uint32_t port_val;
+
+	port_val = get_port(gpio_config);
+	port_val |= pins;
+	set_port(gpio_config, port_val);
+	return 0;
+}
+
+static int gpio_ite_port_clear_bits_raw(struct device *dev,
+					gpio_port_pins_t pins)
+{
+	const struct gpio_ite_cfg *gpio_config = DEV_GPIO_CFG(dev);
+	uint32_t port_val;
+
+	port_val = get_port(gpio_config);
+	port_val &= ~pins;
+	set_port(gpio_config, port_val);
+	return 0;
+}
+
+static int gpio_ite_port_toggle_bits(struct device *dev,
+				     gpio_port_pins_t pins)
+{
+	const struct gpio_ite_cfg *gpio_config = DEV_GPIO_CFG(dev);
+	uint32_t port_val;
+
+	port_val = get_port(gpio_config);
+	port_val ^= pins;
+	set_port(gpio_config, port_val);
+	return 0;
+}
+
+static void gpio_ite_isr(void *arg)
+{
+	int i;
+	struct device *dev = arg;
+	struct gpio_ite_wui *wui_local;
+	const struct gpio_ite_cfg *gpio_config = DEV_GPIO_CFG(dev);
+	int irq = gpio_config->gpio_irq;
+
+	if (gpio_config->reg_addr == GPIO_GPDRB) {
+		for (i = 0; i < 8; i++) {
+			wui_local = (struct gpio_ite_wui *)&GPIO_GPDRB_wui[i];
+			if (irq == wui_local->irq) {
+				SET_MASK(*wui_local->clear_addr,
+					 wui_local->pin);
+			}
+		}
+	}
+	if (gpio_config->reg_addr == GPIO_GPDRF) {
+		for (i = 0; i < 8; i++) {
+			wui_local = (struct gpio_ite_wui *)&GPIO_GPDRF_wui[i];
+			if (irq == wui_local->irq) {
+				SET_MASK(*wui_local->clear_addr,
+					 wui_local->pin);
+			}
+		}
+	}
+}
+
+static int gpio_ite_pin_interrupt_configure(struct device *dev,
+	gpio_pin_t pin, enum gpio_int_mode mode, enum gpio_int_trig trig)
+{
+	int ret = 0;
+	uint32_t g, i;
+	uint8_t both_tri_en = 0;
+	const struct gpio_ite_cfg *gpio_config = DEV_GPIO_CFG(dev);
+	volatile uint8_t *trig_mode;
+	volatile uint8_t *hl_trig;
+
+	g = gpio_config->gpio_irq / 8;
+	i = gpio_config->gpio_irq % 8;
+	trig_mode = reg_ielmr[g];
+	hl_trig = reg_ipolr[g];
+	if (mode & GPIO_INT_MODE_DISABLED) {
+		/* Disables interrupt for a pin. */
+		ite_intc_irq_disable(gpio_config->gpio_irq);
+		return ret;
+	} else if (mode & GPIO_INT_MODE_EDGE) {
+		/* edge trigger */
+		SET_MASK(*trig_mode, BIT(i));
+	} else {
+		/* level trigger */
+		CLEAR_MASK(*trig_mode, BIT(i));
+	}
+	/* both */
+	if ((trig & GPIO_INT_TRIG_LOW) && (trig & GPIO_INT_TRIG_HIGH)) {
+		both_tri_en = 1;
+	} else {
+		if (trig & GPIO_INT_TRIG_LOW) {
+			SET_MASK(*hl_trig, BIT(i));
+		} else if (trig & GPIO_INT_TRIG_HIGH) {
+			CLEAR_MASK(*hl_trig, BIT(i));
+		}
+	}
+	/* set wui , only gpiob gpiof, currently */
+	if (gpio_config->reg_addr == GPIO_GPDRB) {
```

Comment:
This needs a flag in gpio_config as to whether to do this.

Also why are you using GPIO_GPDRB_wui? If it is a macro it should be in all CAPS. But shouldn't you have a pointer in gpio_config abd then use a proper struct? Otherwise this adds to the literal pool